clang-tidy-checks/potential-unbounded-loop-check.diff

```diff
@@ -0,0 +1,3051 @@
++        _gencode_helper("<", "--", t1, t2, trigger=True, swap_comp_order=True),
++        _gencode_helper("<=", "--", t1, t2, trigger=True, swap_comp_order=True),
++    ]
++
++    return "".join(tests)
++
++
++def gencode_eq(t1):
++    """Generates tests using single loop var type
++
++    Arguments:
++        t1  Integer type
++    """
++    tests = [
++        _gencode_helper("<", "++", t1, t1),
++        _gencode_helper("<=", "++", t1, t1),
++        _gencode_helper(">", "--", t1, t1),
++        _gencode_helper(">=", "--", t1, t1),
++    ]
++
++    return "".join(tests)
++
++
++def gen(test):
++    _, types = test
++    n = len(types)
++
++    code = []
++
++    for i in range(n):
++        code.append(gencode_eq(types[i]))
++        for j in range(i + 1, n):
++            code.append(gencode_lt(types[i], types[j]))
++
++    return "".join(code)
++
++
++def main():
++    test_suite = {
++        "signed_int_types": ["int8_t", "int16_t", "int32_t", "int64_t"],
++        "unsigned_int_types": ["uint8_t", "uint16_t", "uint32_t", "uint64_t"],
++    }
++
++    preamble = textwrap.dedent(
++        f"""\
++        // @generated clang-tools-extra/clang-tidy/misc/gen_potential_unbounded_loop_test.py
++        // RUN: %check_clang_tidy %s {CHECK_NAME} %t
++    """
++    )
++
++    code = ["#include<stdint.h>\n"]
++
++    for test in test_suite.items():
++        code.append(gen(test))
++
++    print(f'{preamble} {"".join(code)}')
++
++
++if __name__ == "__main__":
++    main()
+diff --git a/clang-tools-extra/clang-tidy/rename_check.py b/clang-tools-extra/clang-tidy/rename_check.py
+index 4d5311c9a293..2410041fd5d2 100755
+--- a/clang-tools-extra/clang-tidy/rename_check.py
++++ b/clang-tools-extra/clang-tidy/rename_check.py
+@@ -120,16 +120,15 @@ def adapt_cmake(module_path, check_name_camel):
+       if (not file_added) and (cpp_line or cpp_found):
+         cpp_found = True
+         if (line.strip() > cpp_file) or (not cpp_line):
+-          f.write('  ' + cpp_file + '\n')
++          f.write(('  ' + cpp_file + '\n').encode())
+           file_added = True
+-      f.write(line)
++      f.write(line.encode())
+ 
+   return True
+ 
+ # Modifies the module to include the new check.
+ def adapt_module(module_path, module, check_name, check_name_camel):
+-  modulecpp = filter(lambda p: p.lower() == module.lower() + 'tidymodule.cpp',
+-                     os.listdir(module_path))[0]
++  modulecpp = next(filter(lambda p: p.lower() == module.lower() + 'tidymodule.cpp', os.listdir(module_path)))
+   filename = os.path.join(module_path, modulecpp)
+   with open(filename, 'r') as f:
+     lines = f.readlines()
+@@ -149,21 +148,21 @@ def adapt_module(module_path, module, check_name, check_name_camel):
+           header_found = True
+           if match.group(1) > check_name_camel:
+             header_added = True
+-            f.write('#include "' + check_name_camel + '.h"\n')
++            f.write(('#include "' + check_name_camel + '.h"\n').encode())
```

Comment:
Same, why those `.encode()` calls are needed? (And if they are, perhaps they could be added to a separate patch)

Reply:
I agree, it might be better to have a separate patch for this.